pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,277 @@ import (
+	CryptographicEraseDataSanitizationPolicy StorageCohortDataSanitizationPolicy = "CryptographicErase"
+)
+
+type ComponentConditionType string
+
+// These are valid conditions of cohort.
+// In the future, we can add more. The current set of conditions are:
+// CohortComponentReady, CohortComponentNotReady.
+const (
+	// CohortComponentReady means cohort component is healthy and ready to perform its task.
+	CohortComponentReady ComponentConditionType = "Ready"
+	// CohortComponentNotReady means the cohort component is unhealthy and not able to perform its task.
+	CohortComponentNotReady ComponentConditionType = "NotReady"
+	// TODO add more types if necessary
+)
+
+type ComponentConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a component is in the condition.
+// "ConditionFalse" means a component is not in the condition. "ConditionUnknown" means
+// cohort-operator/cohort-manager can't decide if a cohort component is in the condition or not.
+const (
+	ConditionTrue    ComponentConditionStatus = "True"
+	ConditionFalse   ComponentConditionStatus = "False"
+	ConditionUnknown ComponentConditionStatus = "Unknown"
+)
+
+// StorageCohortAccessMode defines various access modes for cohort.
+type StorageCohortAccessMode string
+
+const (
+	// ReadWriteOnce can be mounted in read/write mode to exactly 1 host
+	ReadWriteOnce StorageCohortAccessMode = "ReadWriteOnce"
+	// ReadOnlyMany can be mounted in read-only mode to many hosts
+	ReadOnlyMany StorageCohortAccessMode = "ReadOnlyMany"
+	// ReadWriteMany can be mounted in read/write mode to many hosts
+	ReadWriteMany StorageCohortAccessMode = "ReadWriteMany"
+)
+
+type ProvisioningPolicy string
+
+const (
+	ThickProvisioning ProvisioningPolicy = "thick"
+	ThinProvisioning  ProvisioningPolicy = "thin"
+)
+
+type StorageCohortPhase string
+
+// These are the valid phases of cohort.
+const (
+	// StorageCohortReady means the cohort has been created/added by the system, and has all its components running.
+	StorageCohortReady StorageCohortPhase = "Ready"
+	// StorageCohortNotReady means the cohort has been configured but not reachable.
+	StorageCohortNotReady StorageCohortPhase = "NotReady"
+)
+
+type StorageCohortMultipathingOption string
+
+const (
+	// OnlineActiveMultipathingOption represent active/active configuration
+	OnlineActiveMultipathingOption StorageCohortMultipathingOption = "OnlineActive"
+	// OnlinePassiveMultipathingOption represent active/passive configuration
+	OnlinePassiveMultipathingOption StorageCohortMultipathingOption = "OnlinePassive"
+	// NoneMultipathingOption represent no configuration
+	NoneMultipathingOption StorageCohortMultipathingOption = "None"
+)
+
+type StorageCohortCompressionAlgorithm string
+
+// Compression algorithms
+const (
+	HuffmanCompression StorageCohortCompressionAlgorithm = "huffman"
+)
+
+type StorageCohortAccessProtocol string
+
+const (
```

Comment:
Hmm, if this is about validation, then there are other protocols that need to be included as well. Something like this: https://www.snia.org/sites/default/files/swordfish/draft/v1.2.2/html/Specification/Swordfish_v1.2.2_Specification.html#table_113

Reply:
This part will be moved to the common capabilities structure.